From fd1862906f70d9233e9d667b3b2eab3ab1271e06 Mon Sep 17 00:00:00 2001
From: Meriipu <Meriipu@users.noreply.github.com>
Date: Mon, 14 Dec 2020 06:32:03 +0100
Subject: [PATCH 7/8] Removed from contextmenu: email-image, save-snapshot,
 send-to-device

I do not use them. Countless times have I clicked email image, and
occasionally I have accidentally saved a frame instead of saving the
video itself.

Also remove inspect accessibility from context menu

It only gets in my way when I accidentally click it instead of clicking
inspect element, and it can be accessed from the accessibility tab in
devtools if needed (I have yet to need it enough for it to have its own
entry in the context menu).
---
 browser/base/content/browser-context.inc | 34 ------------------------
 1 file changed, 34 deletions(-)

diff --git a/browser/base/content/browser-context.inc b/browser/base/content/browser-context.inc
index 296045001930..0c0bea0c60ee 100644
--- a/browser/base/content/browser-context.inc
+++ b/browser/base/content/browser-context.inc
@@ -178,9 +178,6 @@
       <menuitem id="context-saveimage"
                 data-l10n-id="main-context-menu-image-save-as"
                 oncommand="gContextMenu.saveMedia();"/>
-      <menuitem id="context-sendimage"
-                data-l10n-id="main-context-menu-image-email"
-                oncommand="gContextMenu.sendMedia();"/>
       <menuitem id="context-setDesktopBackground"
                 data-l10n-id="main-context-menu-image-set-as-background"
                 oncommand="gContextMenu.setDesktopBackground();"/>
@@ -197,15 +194,6 @@
       <menuitem id="context-saveaudio"
                 data-l10n-id="main-context-menu-audio-save-as"
                 oncommand="gContextMenu.saveMedia();"/>
-      <menuitem id="context-video-saveimage"
-                data-l10n-id="main-context-menu-video-image-save-as"
-                oncommand="gContextMenu.saveVideoFrameAsImage();"/>
-      <menuitem id="context-sendvideo"
-                data-l10n-id="main-context-menu-video-email"
-                oncommand="gContextMenu.sendMedia();"/>
-      <menuitem id="context-sendaudio"
-                data-l10n-id="main-context-menu-audio-email"
-                oncommand="gContextMenu.sendMedia();"/>
       <menuitem id="context-ctp-play"
                 data-l10n-id="main-context-menu-plugin-play"
                 oncommand="gContextMenu.playPlugin();"/>
@@ -219,15 +207,6 @@
       <menuitem id="context-pocket"
                 data-l10n-id="main-context-menu-save-to-pocket"
                 oncommand="Pocket.savePage(gContextMenu.browser, gContextMenu.browser.currentURI.spec, gContextMenu.browser.contentTitle);"/>
-      <menuseparator id="context-sep-sendpagetodevice" class="sync-ui-item"
-                     hidden="true"/>
-      <menu id="context-sendpagetodevice"
-            class="sync-ui-item"
-            data-l10n-id="main-context-menu-send-to-device"
-            hidden="true">
-        <menupopup id="context-sendpagetodevice-popup"
-                   onpopupshowing="(() => { gSync.populateSendTabToDevicesMenu(event.target, gBrowser.currentURI.spec, gBrowser.contentTitle); })()"/>
-      </menu>
       <menuseparator id="context-sep-viewbgimage"/>
       <menuitem id="context-viewbgimage"
                 data-l10n-id="main-context-menu-view-background-image"
@@ -287,15 +266,6 @@
                 oncommand="BrowserSearch.loadSearchFromContext(this.searchTerms, this.usePrivate, this.principal, this.csp);"/>
       <menuitem id="context-searchselect-private"
                 oncommand="BrowserSearch.loadSearchFromContext(this.searchTerms, true, this.principal, this.csp);"/>
-      <menuseparator id="context-sep-sendlinktodevice" class="sync-ui-item"
-                     hidden="true"/>
-      <menu id="context-sendlinktodevice"
-            class="sync-ui-item"
-            data-l10n-id="main-context-menu-link-send-to-device"
-            hidden="true">
-        <menupopup id="context-sendlinktodevice-popup"
-                   onpopupshowing="gSync.populateSendTabToDevicesMenu(event.target, gContextMenu.linkURL, gContextMenu.linkTextStr);"/>
-      </menu>
       <menuseparator id="frame-sep"/>
       <menu id="frame" data-l10n-id="main-context-menu-frame">
         <menupopup>
@@ -379,10 +349,6 @@
                 data-l10n-id="main-context-menu-bidi-switch-page"
                 oncommand="gContextMenu.switchPageDirection();"/>
       <menuseparator id="inspect-separator" hidden="true"/>
-      <menuitem id="context-inspect-a11y"
-                hidden="true"
-                data-l10n-id="main-context-menu-inspect-a11y-properties"
-                oncommand="gContextMenu.inspectA11Y();"/>
       <menuitem id="context-inspect"
                 hidden="true"
                 data-l10n-id="main-context-menu-inspect-element"
-- 
2.29.2

